packages/gestalt/src/SideNavigation.js

```diff
@@ -77,7 +78,7 @@ export default function SideNavigation({
     return (
       <SideNavigationProvider
         dismissButton={{
-          accessibilityLabel: dismissButton?.accessibilityLabel,
+          accessibilityLabel: dismissButton?.accessibilityLabel ?? accessibilityDismissButtonLabel,
```

Comment:
Could you clean up this now? dismissButton
<img width="611" alt="Screenshot 2023-06-05 at 3 22 38 PM" src="https://github.com/pinterest/gestalt/assets/10593890/47b06ef1-d84b-4817-a78d-5c8d98db1d3f">

Reply:
DONE, but this isn't going to have quite the same effect. If the provider ever wraps code like this:
![image](https://github.com/pinterest/gestalt/assets/4423381/d545a091-26b3-4419-a6f7-fc19cec4d7a5)
It's always going to set defaults and render the button, instead of having the dismissButton value being null and preventing the button render.

Reply:
Will do in https://jira.pinadmin.com/browse/GESTALT-6329, out of scope for this PR